### PR TITLE
Add author information

### PR DESCRIPTION
**Pull request by fcmsilva**
For the new notebooks, we'll have 2 main changes: authors and categories.
This change adds the authors to the notebooks (defaults to SingleStore), and includes the authors information/image
image

### CODE REVIEW

**Review by kesmit13 — COMMENTED** [review]
If the `authors` key must point to a file in the `authors` directory, we need to have a check added in `resource/nb-check.py` to make sure it exists before the notebook is pushed.

**Review by kesmit13 — COMMENTED** [review]
The check also needs to look for the author image.

**Comment by fcmsilva** [comment]
@kesmit13 Thanks for the feedback, just changed the notebook meta checks (renamed step for clarity), and validated the author entries as well

File: resources/author-check.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+        if 'external' not in meta:
+            error(f'No `external` in `meta` section of {author_path}')
+
+        # Logo is optional, but if defined a corresponding image must exist
+        if 'logo' in meta:
+            logo_id = meta['logo']
+            logo_filename = f'{logo_id}.png'
+            logo_path = os.path.join('common/images/author-images', logo_filename)
+            if not os.path.isfile(logo_path):
```
Comment (tiimgreen) [inline resources/author-check.py:32]: should we also allow image URLs here?

File: resources/author-check.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+def check_author(author_path):
+    print(f'Checking {author_path}...')
+
+    with open(author_path, 'rb') as f:
+        meta = tomllib.load(f)
+
+        if 'name' not in meta:
+            error(f'No `name` in `meta` section of {author_path}')
+
+        if 'description' not in meta:
```
Comment (tiimgreen) [inline resources/author-check.py:21]: nit: I think `title` makes more sense here

File: resources/nb-meta-check.py [hunk 3]
```diff
@@ -26,6 +27,14 @@ def error(msg):
     if 'description' not in meta:
         error(f'No `description` in `meta` section of {f}')
 
+    # Authors must be a non-empty list
+    if (
+        'authors' not in meta
+        or not isinstance(meta['authors'], list)
```
Comment (tiimgreen) [inline resources/nb-meta-check.py:33]: Does the design allow for multiple authors? If not this should be a singular entry

File: notebooks/atlas-and-kai/meta.toml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 [meta]
+authors=["singlestore"]
```
Comment (tiimgreen) [inline notebooks/atlas-and-kai/meta.toml:2]: I'm concerned that doing more of this stuff will make it too tricky to submit a new notebook.

I think we should just have `author_name`, `author_title`, etc. in each `meta.toml` file. It is duplicated yes but we can validate with the checks and most people only add one notebook at a time so not a big deal.

File: resources/nb-meta-check.py (hunk 3 above)
Comment (fcmsilva) [reply resources/nb-meta-check.py:33]: It does, in those cases we use the letter and title of the first author

image

File: resources/author-check.py (hunk 1 above)
Comment (fcmsilva) [reply resources/author-check.py:32]: I prefer to have the images in the folder, seems more organized. We could allow for it to link to CS images for example. But I'm not sure about linking to external images

File: notebooks/atlas-and-kai/meta.toml (hunk 4 above)
Comment (tiimgreen) [reply notebooks/atlas-and-kai/meta.toml:2]: Ah I see, ok yeah let's leave it like this then

File: resources/author-check.py (hunk 1 above)
Comment (fcmsilva) [reply resources/author-check.py:32]: Hmm, maybe I can do something like, if you insert a valid url we assume that's a valid external image. If you just add the filename, we then check if `common/images/author-images/` has that file, would that work?
So, either:
`image="singlestore.png"`, checking if `common/images/author-images/singlestore.png` exists
or
`image=https://externalwebsite.com/image.png`
Would that work?

Comment (tiimgreen) [reply resources/author-check.py:32]: Yeah sounds good

**Review by tiimgreen — APPROVED** [review]

**Pull request merged**